From dc3bcd2da83446fec9e252ac07c5b2840d41298e Mon Sep 17 00:00:00 2001
From: Ankit Agarwal <ankit.agarwal@intel.com>
Date: Thu, 13 Jun 2024 09:31:05 +0530
Subject: [PATCH] Updated Flags to fix build error.

Observed build failure in A15 due to missing flags and due to
inheritance declaration.

Tracked-On: NA
Signed-off-by: Ankit Agarwal <ankit.agarwal@intel.com>
---
 _studio/shared/include/mfx_utils.h | 2 +-
 android/mfx_defs.mk                | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/_studio/shared/include/mfx_utils.h b/_studio/shared/include/mfx_utils.h
index 8394cc48..33bbfe26 100644
--- a/_studio/shared/include/mfx_utils.h
+++ b/_studio/shared/include/mfx_utils.h
@@ -131,7 +131,7 @@ template <class F>
 struct result_of;
 
 template <typename TRes, typename... TArgs>
-struct result_of<TRes(TArgs...)> : std::result_of<TRes(TArgs...)> {};
+struct result_of<TRes(TArgs...)> {};
 
 template <typename TRes, typename... TArgs>
 struct result_of<TRes(*const&)(TArgs...)>
diff --git a/android/mfx_defs.mk b/android/mfx_defs.mk
index fac4bcdd..abfeae31 100644
--- a/android/mfx_defs.mk
+++ b/android/mfx_defs.mk
@@ -21,6 +21,9 @@ MEDIA_VERSION_ALL := $(MEDIA_VERSION).pre$(MEDIA_VERSION_EXTRA)
 MFX_CFLAGS += -DMEDIA_VERSION_STR=\"\\\"${MEDIA_VERSION_ALL}\\\"\"
 
 # Android version preference:
+ifneq ($(filter 15 15.% V% ,$(PLATFORM_VERSION)),)
+  MFX_ANDROID_VERSION:= MFX_V
+endif
 ifneq ($(filter 14 14.% U% ,$(PLATFORM_VERSION)),)
   MFX_ANDROID_VERSION:= MFX_U
 endif
-- 
2.34.1

